python/lsst/pipe/tasks/multiBand.py

```diff
@@ -495,6 +498,10 @@ def runDataRef(self, patchRefList, psfCache=100):
                 filter = patchRef.get(coaddType + "Coadd_filterLabel", immediate=True)
                 filters.append(filter.bandLabel)
                 exposures.append(exposure)
+            # Sort inputs by band to match Gen3 order of inputs
+            exposures = [exposure for _, exposure in sorted(zip(filters, exposures))]
+            patchRefList = [patchRef for _, patchRef in sorted(zip(filters, patchRefList))]
+            filters.sort()
```

Comment:
Why sort the filters *after* sorting the exposures and patchRefs by the filters?

Reply:
They all have to be sorted consistently. The `sorted(zip(list1, list2))` function does the cross-list sorting, so I need to leave them both as they are to begin with, but then the filter list also needs explicit sorting.  That made no sense, so, here:
```
In [36]: filters = ["y", "u", "s", "r", "a"]
In [37]: exposures = [1, 2, 3, 4, 5]
In [38]: exposures = [exposure for _, exposure in sorted(zip(filters, exposures))]
In [39]: exposures
Out[39]: [5, 4, 3, 2, 1] # exposures now ordered in sorted filter order 
In [40]: filters
Out[40]: ['y', 'u', 's', 'r', 'a'] # filters are not, so sort them
In [41]: filters.sort()
In [42]: filters
Out[42]: ['a', 'r', 's', 'u', 'y']  # now they line up properly

# If I sorted the filters first:
In [42]: filters
Out[42]: ['a', 'r', 's', 'u', 'y']
In [43]: exposures = [1, 2, 3, 4, 5]
In [44]: exposures = [exposure for _, exposure in sorted(zip(filters, exposures))]
In [45]: exposures
Out[45]: [1, 2, 3, 4, 5] # Wrong...just the original order!
```
The ordering must be self consistent for the subsequent call to `afwImage.MultibandExposure.fromExposures(filters, exposures)`